stretch_core/stretch_core/joint_trajectory_server.py

```diff
@@ -204,3 +215,57 @@ def success_callback(self, success_str):
         self.result.error_code = self.result.SUCCESSFUL
         self.result.error_string = success_str
         self.goal_handle.succeed()
+
+    def execute_trajectory(self, goal_handle):
+        try:
+            with self.node.robot_stop_lock:
+                # Escape stopped mode to execute trajectory
+                self.node.stop_the_robot = False
+                self.node.robot_mode_rwlock.acquire_read()
+
+            # Process the goal
+            goal = goal_handle.request
+
+            if self.ignore_trajectory_velocities or self.ignore_trajectory_accelerations:
+                for pt in goal.trajectory.points:
+                    if self.ignore_trajectory_velocities:
+                        pt.velocities = []
+                    if self.ignore_trajectory_accelerations:
+                        pt.accelerations = []
+
+            # Print the goal
+            if goal.trajectory.points:
+                dt = to_sec(goal.trajectory.points[-1].time_from_start)
+                n_points = len(goal.trajectory.points)
+            else:
+                dt = 0.0
+                n_points = 0
+
+            n_joints = len(goal.trajectory.joint_names)
+            self.node.get_logger().info(
+                f'New follow_joint_trajectory goal with {n_points} points, {n_joints} joints over {dt} seconds.')
+
+            # TODO: Add the waypoints
+
+            self.node.robot.start_trajectory()
+
+            while rclpy.ok() and self.node.robot.is_trajectory_executing():
+                # TODO: Publish Feedback
+                # TODO: Check Path Tolerances
+                self.trajectory_rate.sleep()
+
+            self.node.robot.stop_trajectory()
+
+            # TODO: Check Goal Tolerances
+
+            goal_handle.succeed()
+            return FollowJointTrajectory.Result(error_code=FollowJointTrajectory.Result.SUCCESSFUL,
+                                                error_string='Achieved all target points.')
+
+        except Exception as e:
+            self.node.robot.stop_trajectory()
+            self.node.get_logger().error(str(traceback.format_exc()))
+            goal_handle.abort()
+            return FollowJointTrajectory.Result(error_code=-10000, error_string=str(e))
```

Comment:
What does -10000 stand for? Can you add a comment here or point out to relevant error?

Reply:
There isn't a code for "unknown error" in [`FollowJointTrajectory`](http://docs.ros.org/en/api/control_msgs/html/action/FollowJointTrajectory.html) so I chose one arbitrarily. I'll add a comment.

Reply:
That would be great, thanks!